src/rgw/rgw_admin.cc

```diff
@@ -6425,26 +6425,39 @@ int main(int argc, const char **argv)
     rgw_bucket bucket;
     RGWBucketInfo bucket_info;
     map<string, bufferlist> attrs;
-    ret = init_bucket(tenant, bucket_name, bucket_id, bucket_info, bucket, &attrs);
+    bool bucket_initable = true;
+    ret = init_bucket(tenant, bucket_name, bucket_id, bucket_info, bucket,
+		      &attrs);
     if (ret < 0) {
-      cerr << "ERROR: could not init bucket: " << cpp_strerror(-ret) << std::endl;
-      return -ret;
+      if (yes_i_really_mean_it) {
+	bucket_initable = false;
+      } else {
+	cerr << "ERROR: could not init bucket: " << cpp_strerror(-ret) <<
+	  "; if you want to cancel the reshard request nonetheless, please "
+	  "use the --yes-i-really-mean-it option" << std::endl;
+	return -ret;
+      }
     }
 
-    RGWBucketReshard br(store, bucket_info, attrs, nullptr /* no callback */);
-    int ret = br.cancel();
-    if (ret < 0) {
-      if (ret == -EBUSY) {
-	cerr << "There is ongoing resharding, please retry after " <<
-	  store->ctx()->_conf.get_val<uint64_t>(
-	    "rgw_reshard_bucket_lock_duration") <<
-	  " seconds " << std::endl;
-      } else {
-	cerr << "Error canceling bucket " << bucket_name <<
-	  " resharding: " << cpp_strerror(-ret) << std::endl;
+    if (bucket_initable) {
+      // we did not encounter an error, so let's work with the bucket
+      RGWBucketReshard br(store, bucket_info, attrs,
+			  nullptr /* no callback */);
+      int ret = br.cancel();
+      if (ret < 0) {
+	if (ret == -EBUSY) {
+	  cerr << "There is ongoing resharding, please retry after " <<
+	    store->ctx()->_conf.get_val<uint64_t>(
+	      "rgw_reshard_bucket_lock_duration") <<
+	    " seconds " << std::endl;
```

Comment:
maybe this could go to the previous line or align with the beginning of the prev. line?

Reply:
Hm seeing this is in other places, most likely it's a tabs vs spaces render on github thing

Reply:
So the emacs directive at the top of the file allows tabs to be used in lieu of 8 spaces. I removed all spaces in this code block and it renders much better in github.